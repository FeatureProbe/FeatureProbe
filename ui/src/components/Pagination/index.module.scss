.pagination {
  margin: 16px 0;
  text-align: right;
  display: flex;
  justify-content: space-between;
  padding-left: 24px;
}

.total {
  color: #74788d;
  font-size: 12px;
}

.total-count {
  font-size: 14px;
}