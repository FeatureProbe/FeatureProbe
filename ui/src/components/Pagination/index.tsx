import { SyntheticEvent } from 'react';
import { FormattedMessage } from 'react-intl';
import { Pagination, PaginationProps } from 'semantic-ui-react';
import Icon from 'components/Icon';
import styles from './index.module.scss';

interface IPagination {
  pageIndex: number;
  totalPages: number;
}

interface IProps {
  text?: string;
  total?: number;
  hideTotal?: boolean;
  pagination: IPagination;
  handlePageChange(e: SyntheticEvent, data: PaginationProps): void;
}

const CustomPagination = (props: IProps) => {
  const { total, text, hideTotal, pagination, handlePageChange } = props;

  return (
    <div className={styles.pagination}>
      {
        hideTotal ? <div className={styles['total']}></div> : (
          <div className={styles['total']}>
            <span className={styles['total-count']}>{total} </span>
            { text || <FormattedMessage id='common.total' /> }
          </div>
        )
      }
      {
        pagination.totalPages > 1 && (
          <Pagination 
            activePage={pagination.pageIndex} 
            totalPages={pagination.totalPages} 
            onPageChange={handlePageChange}
            firstItem={null}
            lastItem={null}
            prevItem={{
              content: (<Icon type='angle-left' />)
            }}
            nextItem={{
              content: (<Icon type='angle-right' />)
            }}
          />
        )
      }
    </div>
  );
};

export default CustomPagination;