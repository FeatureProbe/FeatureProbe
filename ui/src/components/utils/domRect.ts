export class DOMRect {
    x: number;
    width: number;
    y: number;
    height: number;
    constructor(x: number, y: number, width: number, height: number) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}