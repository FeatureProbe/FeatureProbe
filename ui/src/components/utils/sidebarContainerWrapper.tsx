import { SidebarContainer } from 'layout/hooks';

export default SidebarContainer.Provider;
