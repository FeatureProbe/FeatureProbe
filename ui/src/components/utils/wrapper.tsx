import en_US from '../../locales/en-US.json';
import { IntlProvider } from 'react-intl';

export const IntlWrapper: React.FC = ({ children }) => {
  return (
    <IntlProvider locale="en" messages={en_US}>{children}</IntlProvider>
  );
};
