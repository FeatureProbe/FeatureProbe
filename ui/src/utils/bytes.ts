export function getBytes(str: string) {
  let bytes = str.length,
    i = 0;
  for (; i < bytes; i++) if (str.charCodeAt(i) > 255) bytes++;

  return bytes;
}

const MB = 1048576;
export const MAX_SIZE = MB;