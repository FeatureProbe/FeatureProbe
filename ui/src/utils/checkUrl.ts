function escapeStringRegexp(string: string): string {
  if (typeof string !== 'string') {
    throw new TypeError('Expected a string');
  }

  return string
    .replace(/[|\\{}()[\]^$+*?.]/g, '\\$&')
    .replace(/-/g, '\\x2d');
}

export function matchUrl(matcher: string, originUrl: string, targetUrl: string): boolean {
  let href = '';
  let hash = '';
  let search = '';

  try {
    const fullUrl = new URL(targetUrl, '');
    href = fullUrl.href;
    hash = fullUrl.hash;
    search = fullUrl.search;
  } catch (error) {
    href = targetUrl;
  }

  let regex;
  let testUrl;

  switch (matcher) {
    case 'EXACT':
      testUrl = href;
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SIMPLE':
      testUrl = href.replace(hash, '').replace(search, '');
      regex = new RegExp('^' + escapeStringRegexp(originUrl) + '/?$');
      break;
    case 'SUBSTRING':
      testUrl = href.replace(search, '');
      regex = new RegExp('.*' + escapeStringRegexp(originUrl) + '.*$');
      break;
    case 'REGULAR':
      testUrl = href.replace(search, '');
      regex = new RegExp(originUrl);
      break;
    default:
      return false;
  }

  return regex.test(testUrl);
}