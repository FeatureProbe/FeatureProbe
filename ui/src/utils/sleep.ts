const sleep = (delay: number) => {
  return new Promise((res) => {
    setTimeout(() => {
      res('ok');
    }, delay);
  });
};

export default sleep;
