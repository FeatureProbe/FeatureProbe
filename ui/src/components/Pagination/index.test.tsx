import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import userEvent from '@testing-library/user-event';
import Pagination from '.';
import sleep from 'utils/sleep';
import { IntlWrapper } from 'components/utils/wrapper';

it('Pagination snapshot', (done) => {
  (async () => {
    const { asFragment } = render(
      <Pagination 
        handlePageChange={jest.fn()}
        pagination={{
          pageIndex: 0,
          totalPages: 10
        }}
      />
    , { wrapper: IntlWrapper});
    expect(asFragment()).toMatchSnapshot();
    done();
  })();
});

test('Pagination hideTotal', (done) => {
  (async () => {
    render(
      <Pagination 
        handlePageChange={jest.fn()}
        hideTotal={true}
        total={100}
        pagination={{
          pageIndex: 0,
          totalPages: 10
        }}
      />
    , { wrapper: IntlWrapper});

    const ele = document.getElementsByClassName('total-count')[0];
    expect(ele).toBeUndefined();

    done();
  })();
});

test('Pagination Click Test', (done) => {
  (async () => {
    const mockHandleChange = jest.fn();

    render(
      <Pagination 
        handlePageChange={mockHandleChange}
        hideTotal={false}
        total={100}
        pagination={{
          pageIndex: 0,
          totalPages: 10
        }}
      />
    , { wrapper: IntlWrapper});

    // Click next page icon
    await userEvent.click(document.getElementsByClassName('icon-angle-right')[0]);
    expect(mockHandleChange).toBeCalled();

    await sleep(1000);

    // Click page 10
    await userEvent.click(screen.getByText('10'));
    expect(mockHandleChange).toBeCalled();

    done();
  })();
});
