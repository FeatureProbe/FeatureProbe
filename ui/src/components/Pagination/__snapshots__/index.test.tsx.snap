// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination snapshot 1`] = `
<DocumentFragment>
  <div
    class="pagination"
  >
    <div
      class="total"
    >
      <span
        class="total-count"
      >
         
      </span>
       in total
    </div>
    <div
      aria-label="Pagination Navigation"
      class="ui pagination menu"
      role="navigation"
    >
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="prevItem"
        value="1"
      >
        <span
          class="iconfont icon-angle-left undefined"
          type="angle-left"
        />
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="pageItem"
        value="1"
      >
        1
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="pageItem"
        value="2"
      >
        2
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="pageItem"
        value="3"
      >
        3
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="pageItem"
        value="4"
      >
        4
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="pageItem"
        value="5"
      >
        5
      </a>
      <a
        aria-current="false"
        aria-disabled="true"
        class="item"
        tabindex="-1"
        type="ellipsisItem"
        value="6"
      >
        ...
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="pageItem"
        value="10"
      >
        10
      </a>
      <a
        aria-current="false"
        aria-disabled="false"
        class="item"
        tabindex="0"
        type="nextItem"
        value="1"
      >
        <span
          class="iconfont icon-angle-right undefined"
          type="angle-right"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;
